main.js

```diff
@@ -62,5 +62,37 @@ function init(options, dev) {
       app.ports['deleteSession'].send(sessionId);
     }
   });
+// thanks to https://github.com/phylor/elm-image-upload/blob/master/index.html
```

Comment:
do we need more ports to do this? There is this [library](https://github.com/simonh1000/file-reader) that handles file upload without ports. Example here http://simonh1000.github.io/2016/12/elm-s3-uploads/

Reply:
From what I've read, using libraries native code seems to have pitfalls as well so no need actually!